logbook-api/src/main/java/org/zalando/logbook/api/RequestURI.java

```diff
@@ -7,35 +7,35 @@
 import java.util.Set;
 
 import static java.util.Arrays.asList;
-import static org.zalando.logbook.RequestURI.Component.AUTHORITY;
-import static org.zalando.logbook.RequestURI.Component.PATH;
-import static org.zalando.logbook.RequestURI.Component.QUERY;
-import static org.zalando.logbook.RequestURI.Component.SCHEME;
+import static org.zalando.logbook.api.RequestURI.Component.AUTHORITY;
+import static org.zalando.logbook.api.RequestURI.Component.PATH;
+import static org.zalando.logbook.api.RequestURI.Component.QUERY;
+import static org.zalando.logbook.api.RequestURI.Component.SCHEME;
 
 @UtilityClass
-final class RequestURI {
+public final class RequestURI {
```

Comment:
Looks like this class and `LogbookFactory` are becoming public. Not sure if we should be concerned about it. I wonder if it would be possible to hide internal classes like this one behind the java 9 modules. Would it be compatible with java 8 somehow? Does it make sense to do it?

Reply:
We can use Java 9 modules to expose classes of one package (e.g., `org.zalando.logbook.internal`) to specific modules only (e.g., `core` module). This still requires the classes to be **public**. Also, it won't work with Java 8 as it does not recognize modules.

More explanation from ChatGPT 🙂

![image](https://user-images.githubusercontent.com/3616518/230093229-82d2c7bb-d5dd-4ef6-8705-3cb1a37c0020.png)